src/Compiler/MacroProcessor.cpp

```diff
@@ -144,11 +215,8 @@ namespace Ark::internal
         }
     }
 
-    void MacroProcessor::execute(Node& node)
-    {
-        static std::function<void(const std::unordered_map<std::string, Node>&, Node&, Node*)> apply_to =
-        [this](const std::unordered_map<std::string, Node>& map, Node& target, Node* parent) {
-            if (target.nodeType() == NodeType::Symbol)
+    void MacroProcessor::apply_to(const std::unordered_map<std::string, Node>& map, Node& target, Node* parent){
```

Comment:
A bit of coding style, the `{` should be on its own line, and the rest of the code has too much indentations (except for line 219 which is correct)